x-pack/plugins/fleet/server/services/api_keys/enrollment_api_key_so.ts

```diff
@@ -112,7 +113,31 @@ export async function generateEnrollmentAPIKey(
   }
   const agentPolicyId =
     data.agentPolicyId ?? (await agentPolicyService.getDefaultAgentPolicyId(soClient));
+
+  if (providedKeyName) {
+    const { items } = await listEnrollmentApiKeys(soClient, {
+      kuery: `${ENROLLMENT_API_KEYS_SAVED_OBJECT_TYPE}.policy_id:${agentPolicyId} and ${ENROLLMENT_API_KEYS_SAVED_OBJECT_TYPE}.name:${providedKeyName.replace(
+        / /g,
+        '*'
```

Comment:
SavedObjects doesn't seem to support escaped spaces in kuery filters, so instead spaces are converted to `*`.

Not an ideal solution, as it could result in an edge case where a user can't create a token named `ab de` if they already have one named `abcde`. However, the Fleet Server implementation supports escaping spaces in the filter perfectly fine, so this might be okay to push assuming SavedObjects is going away eventually.